samples/v1/types_config.go

```diff
@@ -0,0 +1,215 @@
+
+const (
+	// ImportCredentialsExist represents the state of any credentials specified by
+	// the SamplesRegistry field in the Spec.
+	ImportCredentialsExist ConfigConditionType = "ImportCredentialsExist"
+	// SamplesExist represents whether an incoming Config has been successfully
+	// processed or not all, or whether the last Config to come in has been
+	// successfully processed.
+	SamplesExist ConfigConditionType = "SamplesExist"
+	// ConfigurationValid represents whether the latest Config to come in
+	// tried to make a support configuration change.  Currently, changes to the
+	// InstallType and Architectures list after initial processing is not allowed.
+	ConfigurationValid ConfigConditionType = "ConfigurationValid"
+	// ImageChangesInProgress represents the state between where the samples operator has
+	// started updating the imagestreams and when the spec and status generations for each
+	// tag match.  The list of imagestreams that are still in progress will be stored
+	// in the Reason field of the condition.  The Reason field being empty corresponds
+	// with this condition being marked true.
+	ImageChangesInProgress ConfigConditionType = "ImageChangesInProgress"
+	// RemovePending represents whether the Config Spec ManagementState
+	// has been set to Removed, but we have not completed the deletion of the
+	// samples, pull secret, etc. and set the Config Spec ManagementState to Removed.
+	// Also note, while a samples creation/update cycle is still in progress, and ImageChagesInProgress
+	// is True, the operator will not initiate the deletions, as we
+	// do not want the create/updates and deletes of the samples to be occurring in parallel.
+	// So the actual Removed processing will be initated only after ImageChangesInProgress is set
+	// to false.  Once the deletions are done, and the Status ManagementState is Removed, this
+	// condition is set back to False.  Lastly, when this condition is set to True, the
+	// ClusterOperator Progressing condition will be set to True.
+	RemovePending ConfigConditionType = "RemovePending"
+	// MigrationInProgress represents the special case where the operator is running off of
+	// a new version of its image, and samples are deployed of a previous version.  This condition
+	// facilitates the maintenance of this operator's ClusterOperator object.
+	MigrationInProgress ConfigConditionType = "MigrationInProgress"
+	// ImportImageErrorsExist registers any image import failures, separate from ImageChangeInProgress,
+	// so that we can a) indicate a problem to the ClusterOperator status, b) mark the current
+	// change cycle as complete in both ClusterOperator and Config; retry on import will
+	// occur by the next relist interval if it was an intermittent issue;
+	ImportImageErrorsExist ConfigConditionType = "ImportImageErrorsExist"
+	// numConfigConditionType is a helper constant that captures the number possible conditions
+	// defined above in this const block
+	numconfigConditionType = 7
```

Comment:
@gabemontero now that we've moved this to api, do we need it any more? It isn't exported, so I don't see what if anything could use it.

Reply:
yep I should remove that thanks